graylog2-server/src/main/java/org/graylog2/indexer/messages/IndexAndMessage.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.indexer.messages;
+
+import org.graylog2.plugin.Message;
+
+public class IndexAndMessage {
+    public String indexName;
+    public Message message;
+
+    public IndexAndMessage(String indexName, Message message) {
```

Comment:
I am fine with using an immutable map entry. Maybe we should use a `Map.Entry<IndexSet, Message>` instead of `Map.Entry<String, Message>`. Since we only have `#getKey()` and `#getValue()` it would be hard to see what is in the key. :wink: